python/rikai/viz.py

```diff
@@ -0,0 +1,150 @@
+#  Copyright 2021 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Optional
+
+import numpy as np
+
+from rikai.mixin import Displayable, Drawable
+
+__all__ = ["Style"]
+
+
+class Style(Drawable):
+    """Styling a drawable-component.
+
+    Examples
+    --------
+
+    >>> from rikai.viz import Style
+    >>> from rikai.types import Box2d, Image
+    ...
+    >>> img = Image(uri="s3://....")
+    >>> bbox1, bbox2 = Box2d(1, 2, 3, 4), Box2d(3, 4, 5, 6)
+    >>> bbox_style = Style(color="yellow", width=4)
+    >>> image | bbox_style(bbox1) | bbox_style(bbox2)
+    """
+
+    def __init__(self, **kwarg):
+        self.kwargs = kwarg
+        self.inner = None  # type: Optional[Drawable]
+
+    def __repr__(self):
+        return f"style({self.kwargs})"
+
+    def __call__(self, inner: Drawable) -> Drawable:
+        # Make a copy of Style so the same style can be applied
+        # to multiple drawables
+        s = Style(**self.kwargs)
+        s.inner = inner
+        return s
+
+    def render(self, render: Render, **kwargs):
```

Comment:
when would the user need to supply kwargs in the constructor and also in the render call?

Reply:
i should make this a private/protected methods tho. This method is not user-facing interface.

Reply:
Renamed this method to `_render(..)`